api/pkg/apis/projectcalico/v3/felixconfig.go

```diff
@@ -648,6 +648,30 @@ type FelixConfigurationSpec struct {
 	// WindowsManageFirewallRules configures whether or not Felix will program Windows Firewall rules. (to allow inbound access to its own metrics ports) [Default: Disabled]
 	// +optional
 	WindowsManageFirewallRules *WindowsManageFirewallRulesMode `json:"windowsManageFirewallRules,omitempty" validate:"omitempty,oneof=Enabled Disabled"`
+
+	// GoGCThreshold Sets the Go runtime's garbage collection threshold.  I.e. the percentage that the heap is
+	// allowed to grow before garbage collection is triggered.  In general, doubling the value halves the CPU time
+	// spent doing GC, but it also increases memory usage by 50%.  A special value of -1 can be used
```

Comment:
I think I'm right in saying doubling the value does halve the GC CPU time regardless of previous value, but for memory it only increases by 50% when going from 100 to 200.

Peak memory usage is much more important than average because the best practice is to set memoryRequest=memoryLimit, so I think we should offer feedback using that.

I think a reasonable simplification would be to say "doubling the value halves the CPU time spent doing GC, but GC memory overhead is proportional to the value" maybe?

Reply:
I think your wording is better, I struggled to come up with something clear and concise :+1: